internal/controller/components/dashboard/dashboard_support_test.go

```diff
@@ -0,0 +1,712 @@
+		callCount int
+	}{
+		{
+			name:      "SingleCall",
+			callCount: 1,
+		},
+		{
+			name:      "MultipleCalls",
+			callCount: 5,
+		},
+		{
+			name:      "Stability",
+			callCount: 100,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewWithT(t)
+
+			// Test that the function returns valid component names
+			names := make([]string, tt.callCount)
+			for i := range tt.callCount {
+				names[i] = dashboard.ComputeComponentName()
+			}
+
+			// All calls should return non-empty names
+			for i, name := range names {
+				g.Expect(name).ShouldNot(BeEmpty(), "Call %d should return non-empty name", i)
+				g.Expect(name).Should(Or(
+					Equal(dashboard.LegacyComponentNameUpstream),
+					Equal(dashboard.LegacyComponentNameDownstream),
+				), "Call %d should return valid component name", i)
+				g.Expect(len(name)).Should(BeNumerically("<", 50), "Call %d should return reasonable length name", i)
+			}
+
+			// All calls should return identical results (determinism)
+			if tt.callCount > 1 {
+				for i := 1; i < len(names); i++ {
+					g.Expect(names[i]).Should(Equal(names[0]), "All calls should return identical results")
+				}
+			}
+		})
+	}
+}
+
+func TestInit(t *testing.T) {
+	g := NewWithT(t)
+
+	handler := &dashboard.ComponentHandler{}
+
+	// Test successful initialization for different platforms
+	platforms := []common.Platform{
+		cluster.OpenDataHub,
+		cluster.SelfManagedRhoai,
+		cluster.ManagedRhoai,
+	}
+
+	for _, platform := range platforms {
+		t.Run(string(platform), func(t *testing.T) {
+			err := handler.Init(platform)
+			g.Expect(err).ShouldNot(HaveOccurred())
+		})
+	}
+}
+
+func TestSectionTitleMapping(t *testing.T) {
+	g := NewWithT(t)
+
+	expectedMappings := map[common.Platform]string{
+		cluster.SelfManagedRhoai: "OpenShift Self Managed Services",
+		cluster.ManagedRhoai:     "OpenShift Managed Services",
+		cluster.OpenDataHub:      "OpenShift Open Data Hub",
+	}
+
+	for platform, expectedTitle := range expectedMappings {
+		g.Expect(dashboard.SectionTitle[platform]).Should(Equal(expectedTitle))
+	}
+}
+
+func TestBaseConsoleURLMapping(t *testing.T) {
+	g := NewWithT(t)
+
+	expectedMappings := map[common.Platform]string{
+		cluster.SelfManagedRhoai: "https://rhods-dashboard-",
+		cluster.ManagedRhoai:     "https://rhods-dashboard-",
+		cluster.OpenDataHub:      "https://odh-dashboard-",
+	}
+
+	for platform, expectedURL := range expectedMappings {
+		g.Expect(dashboard.BaseConsoleURL[platform]).Should(Equal(expectedURL))
+	}
+}
+
+func TestOverlaysSourcePathsMapping(t *testing.T) {
+	g := NewWithT(t)
+
+	expectedMappings := map[common.Platform]string{
+		cluster.SelfManagedRhoai: "/rhoai/onprem",
+		cluster.ManagedRhoai:     "/rhoai/addon",
+		cluster.OpenDataHub:      "/odh",
+	}
+
+	for platform, expectedPath := range expectedMappings {
+		g.Expect(dashboard.OverlaysSourcePaths[platform]).Should(Equal(expectedPath))
+	}
+}
+
+func TestImagesMap(t *testing.T) {
```

Comment:
We are testing the map here, I'd avoid that

Reply:
why?